lib/temporal/worker.rb

```diff
@@ -0,0 +1,72 @@
+require 'async'
+require 'temporal/bridge'
+require 'temporal/data_converter'
+require 'temporal/failure_converter'
+require 'temporal/payload_converter'
+require 'temporal/runtime'
+require 'temporal/worker/activity'
+require 'temporal/worker/thread_pool_executor'
+
+module Temporal
+  class Worker
+    # TODO: Add worker interceptors
+    def initialize(
+      connection,
+      namespace,
+      task_queue,
+      payload_converter: Temporal::PayloadConverter::DEFAULT,
+      payload_codecs: [],
+      failure_converter: Temporal::FailureConverter::DEFAULT,
+      activity_executor: nil,
+      max_concurrent_activities: 100
+    )
+      @running = false
+      @mutex = Mutex.new
+      @runtime = Temporal::Runtime.instance
+      activity_executor ||= ThreadPoolExecutor.new(max_concurrent_activities)
+      data_converter = DataConverter.new(
```

Comment:
If data converter is a concept, why not just accept that instead of the pieces to make it?

Reply:
I see data converter is a helper class for accessing Payload Convert, Codes and Failure Converter, not something I'd like to expose to the user

Reply:
If it's a helper for you, one might expect it could be a helper for them. Why not expose it? How are we hiding it? If it's nothing more than three collected properties, it seems reasonable to let them pass it around as one as we do.

Reply:
I mean they can use it if they want to (we're not hiding it in any special way), I just think that forcing people to use it (when initialising a Client or Worker) will only make it a bit more difficult for them

Reply:
Most won't use it when initializing, most need it when converting. If I am a user and have a raw payload from some API call I made (maybe I'm listing events or something), how do I convert it to an object like my workflow or activity would use? Same for failures. It is a bit rough to expect users to know they have to go through two things (codec + converter) when we get this nice helper to ourselves.

In every other SDK people can easily just configure/use data converters. We shouldn't make Ruby users suffer unique inconsistency without a very good reason. We're gonna have docs like "in every SDK except Ruby you configure a data converter on your client, in Ruby you configure these three things".

Reply:
Right, my point is that if I want to just add codec, it's easier to do:

```ruby
Temporal::Worker.new(..., payload_codecs: [MyCodec.new])
```

Than:

```ruby
data_converter = Temporal::DataConverter.new(payload_codecs: [MyCodec.new])
Temporal::Worker.new(..., data_converter: data_converter)
```

There's simply more API you need to interact with. Now the first option does not restrict you from using `Temporal::DataConverter.new(payload_codecs: [MyCodec.new])` when you need to, it's just a matter of what's more convenient at the point of Client/Worker initialization. I don't really buy that users will be interacting with raw payloads (maybe in some rare cases), so optimising for that while making a more obvious use case harder seems like a poor design decision.

However if that's what other SDKs do, I'm happy to make this change in Ruby as well — #75

Reply:
"I want to just add codec" is less frequent than "I just want to use the data converter" I think. And also less consistent with other SDKs.

> I don't really buy that users will be interacting with raw payloads 

It's more often than you think, and I'd argue more often than changing codec